src/roles/plugin.ts

```diff
@@ -7,15 +7,16 @@ import type StateCore from "markdown-it/lib/rules_core/state_core"
 import type StateInline from "markdown-it/lib/rules_inline/state_inline"
 import type Token from "markdown-it/lib/token"
 import { Role } from "./main"
+import { inlineMathRenderer, IOptions as IMathRoleOptions } from "./math"
 
 /** Allowed options for directive plugin */
-export interface IOptions {
+export interface IOptions extends IMathRoleOptions {
```

Comment:
I don't think this is the correct way to go; having inheritance add role based options (or indeed any options). We need to come up with a better way of doing this

Reply:
Off hand, I would suggest something like:

```typescript
export interface IOptions {
  suboptions: {
    math: {
      renderer?: (content: string, opts: { displayMode: boolean }) => string
    }
  }
}
```

Reply:
I have adressed this!